File: COMMUTE.c
#include<stdio.h>
int main()
{
    int i,n,T,j,time;
    scanf("%d",&T);
    for(i=0;i<T;i++)
    {
        scanf("%d",&n);
        int x[n],t[n],d[n];
        for(j=0;j<n;j++)
        scanf("%d %d %d",&x[j],&t[j],&d[j]);
        time=0;
        for(j=0;j<n;j++)
        {
        if(time<=x[j])
        time=x[j]+t[j];
        else if((time-x[j])%d[j]==0)
        time=x[j]+((time-x[j])/d[j])*d[j]+t[j];
        else if((time-x[j])%d[j]!=0)
        time=x[j]+((time-x[j])/d[j])*d[j]+t[j]+d[j];
        }
        printf("%d\n",time);
    }
    return 0;
}

File: commute.py
T=int(input())
while T>0:
    n=int(input())
    j=0
    x=[]
    t=[]
    d=[]
    while j<n:
        a,b,c=map(int,input().split(" "))
        x.append(a),t.append(b),d.append(c)
        j+=1
    time=0
    i=0
    while i<n:
        if(time-x[i]<=0):
            time=x[i]+t[i]
        elif((time-x[i])%d[i]==0):
            time=x[i]+((time-x[i])//d[i])*d[i]+t[i]
        elif((time-x[i])%d[i]!=0):
            time=x[i]+((time-x[i])//d[i])*d[i]+t[i]+d[i]
        i+=1
    print(time)
    T-=1
